### PR TITLE
App forms - Blog app 🎲

### PR DESCRIPTION
**Pull request by evansnyamekye**
- Create a method called `current_user` in your ApplicationController to make current user data available to all controllers.
   - [ ]  It will return the first user from the database.
   - [ ] Don't worry about proper authentication at this time. You will add it in the upcoming 
      projects.
- Create forms to perform the following functions:
    - [ ] Creates a Post on behalf of the `current_user` (use the method that you created in your ApplicationController).
    - [ ] Create a comment on behalf of the `current_user` (use the method that you created in your 
    ApplicationController).
- Allow Users to add likes to Posts.

Hello **@Reviewer**,
- This is Link: https://github.com/evansnyamekye/blog-app/pull/8 to previous reviews.

### CODE REVIEW

File: app/views/comments/new.html.erb [hunk 1]
```diff
@@ -1,5 +1,5 @@
 <h3 style="color: #336699; margin: 12px 50px">Create a new comment💬</h3>
-<%= form_with model: [@user, @post, @comment], url: user_post_comments_path(@user, @post), method: :post do |f| %>
+<%= form_with model: [@user, @post, @comment], url: user_post_comments_path(user_id:2, post_id:4), method: :post do |f| %>
   <div class="inputs">
     <%= f.label :text, style: "font-weight: bold;" %>
     <%= f.text_area :text, style: "width: 230px; height: 80px; padding: 10px" %>
```
Comment (mckent05) [inline app/views/comments/new.html.erb]: - Good job done so far, but I have this error when trying to create a new comment, please fix this 🙏
rergerwerfw

**Review by mckent05 — CHANGES_REQUESTED** [review]
Hi @evansnyamekye, 

# STATUS: CHANGES REQUIRED ♻️ ♻️ 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
almost there

## Highlights
- All linter checks are passing ✔️ ✔️ 
- Your PR is professional 👍 👍 
- Good job creating forms for your application 👍 👍 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/controllers/comments_controller.rb [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    @comment = Comment.new
+    @post = Post.find(params[:post_id])
+  end
+
+  def create
+    @comment = @post.comments.build(comment_params.merge(author_id: current_user.id))
+
+    if @comment.save
+      flash[:success] = 'Comment saved successfully'
+      redirect_back_or_default(user_post_comments_path(@user, @post))
```
Comment (Whoistolu) [inline app/controllers/comments_controller.rb:15]: - [ ] [REQUIRED] Great efforts @evansnyamekye ,
However, I am unable to create a comment at the moment. The issue is related to the redirect_back_or_default method. The error message indicates that the :user_id key is missing when trying to generate a URL in the create action of the CommentsController.
Screenshot 2023-12-15 at 15 27 38

### To resolve this issue
- The @user variable is not defined in your create action, and that's why the :user_id key is missing. You should use current_user instead

``` ruby
redirect_to new_user_post_comment_path(current_user, @post)
```

**Review by Whoistolu — CHANGES_REQUESTED** [review]
Hi @evansnyamekye 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️

- [ ]  _Check the inline comments under the review for sections to improve._

### Optional suggestions
Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better.

### N/A

Cheers and Happy coding!

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @whoistolu in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/controllers/comments_controller.rb (hunk 2 above)
Comment (evansnyamekye) [reply app/controllers/comments_controller.rb:15]: Hi, @Whoistolu. The issue with the comment has been resolved. kindly find below a video.  

https://github.com/evansnyamekye/blog-app/assets/60971417/c61e36b2-dc36-4582-aac4-19e9892822f0

**Review by OyePriscilla — APPROVED** [review]
Hi @evansnyamekye 👏 👏, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights
Comment saved succesfully ✔️ 

CAROL

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**